From f86ed770b5396cd510660fb455415ac92083c7f6 Mon Sep 17 00:00:00 2001
From: Arnaldo Carvalho de Melo <acme@redhat.com>
Date: Tue, 27 Feb 2024 10:19:45 -0300
Subject: [PATCH] dtagnames: cus__load_files() requires a non NULL conf_load

Since ff7bd7083f36db80 ("core: Allow sizing the loader hash table")
dwarf__load_files(), called by cus__load_files() needs a non-NULL
conf_load argument, add one to avoid a core dump.

With the fix:

  $ dtagnames vmlinux | sort | uniq -c | sort -nr
  8793952 member
  2214363 formal_parameter
  1707497 pointer_type
  1397044 structure_type
  1101747 enumerator
   819003 variable
   671032 subroutine_type
   441683 inlined_subroutine
   432020 lexical_block
   340042 subprogram
   335456 typedef
   264541 array_type
   208998 const_type
   194379 union_type
   132403 enumeration_type
    41631 base_type
    29098 label
     4941 volatile_type
        4 restrict_type
  $

J B reported this for syscse, which led me to do a quick audit to find
this other case.

Reported-by: J B <jb.1234abcd@gmail.com>
Link: https://lore.kernel.org/dwarves/CAJXMFhFbv5MccigKvmQt9MhYwKv2iyigEG3Bhs64BZ=PMWc0WQ@mail.gmail.com
Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
(cherry picked from commit f86ed770b5396cd510660fb455415ac92083c7f6)
Upstream: https://git.kernel.org/pub/scm/devel/pahole/pahole.git/commit/?id=f86ed770b5396cd510660fb455415ac92083c7f6
---
 dtagnames.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/dtagnames.c b/dtagnames.c
index 343f055d..8bec2cd0 100644
--- a/dtagnames.c
+++ b/dtagnames.c
@@ -11,6 +11,12 @@
 #include "dwarves.h"
 #include "dutil.h"
 
+static struct conf_fprintf conf;
+
+static struct conf_load conf_load = {
+	.conf_fprintf = &conf,
+};
+
 static int class__tag_name(struct tag *tag, struct cu *cu __maybe_unused,
 			   void *cookie __maybe_unused)
 {
@@ -40,7 +46,7 @@ int main(int argc __maybe_unused, char *argv[])
 	}
 	dwarves__resolve_cacheline_size(NULL, 0);
 
-	err = cus__load_files(cus, NULL, argv + 1);
+	err = cus__load_files(cus, &conf_load, argv + 1);
 	if (err != 0) {
 		cus__fprintf_load_files_err(cus, "dtagnames", argv + 1, err, stderr);
 		goto out;
